lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -5901,7 +5901,8 @@ ActorReferenceResult ActorReferenceResult::forReference(
       declIsolation.isGlobalActor()) {
     auto *init = dyn_cast<ConstructorDecl>(fromDC);
     auto *decl = declRef.getDecl();
-    if (init && init->isDesignatedInit() && isStoredProperty(decl)) {
+    if (init && init->isDesignatedInit() && isStoredProperty(decl) &&
```

Comment:
It'd be better to check directly whether the reference is on `self`, but that requires plumbing a base expression through in a couple places. I'll do that separately.